Cache/Main.lean

```diff
@@ -11,31 +11,46 @@ Usage: cache [COMMAND]
 
 Commands:
   # No priviledge required
-  get       Download linked files missing on the local cache and decompress
-  get!      Download all linked files and decompress
-  mk        Compress non-compressed build files into the local cache
-  mk!       Compress build files into the local cache (no skipping)
-  unpack    Decompress linked already downloaded files
-  clean     Delete non-linked files
-  clean!    Delete everything on the local cache
+  get  [ARGS]  Download linked files missing on the local cache and decompress
+  get! [ARGS]  Download all linked files and decompress
+  mk           Compress non-compressed build files into the local cache
+  mk!          Compress build files into the local cache (no skipping)
+  unpack       Decompress linked already downloaded files
+  clean        Delete non-linked files
+  clean!       Delete everything on the local cache
 
   # Privilege required
-  put       Run 'mk' then upload linked files missing on the server
-  put!      Run 'mk' then upload all linked files
-  commit    Write a commit on the server
-  commit!   (Over)write a commit on the server
-  collect   TODO
+  put          Run 'mk' then upload linked files missing on the server
+  put!         Run 'mk' then upload all linked files
+  commit       Write a commit on the server
+  commit!      Overwrite a commit on the server
+  collect      TODO
 
 * Linked files refer to local cache files with corresponding Lean sources
-* Commands ending with '!' should be used manually, when hot-fixes are needed"
+* Commands ending with '!' should be used manually, when hot-fixes are needed
+
+# The arguments for 'get' and 'get!'
+
+'get' and 'get!' can process list of paths containing at most one wildcard each,
+allowing the user to be more specific about what should be downloaded. Example:
+
+$ lake exe cache get Mathlib/Algebra/Field/* Mathlib/Data *Conjugate.lean
```

Comment:
We don't need to specify the `.lean` extension in `lake build SomeFile`, right? Can this/is this already the case here?
```suggestion
$ lake exe cache get Mathlib/Algebra/Field/* Mathlib/Data *Conjugate
```

Reply:
Not in this case. We can't assume that the user always intend to implicitly add `.lean` at the end of the pattern. There's no `Mathlib/Data.lean`, for example.

Reply:
My mental model is that the `.lean` is not part of the file name. So what I would do is make your code `*.lean` to each pattern.

Reply:
!! `.lean` is definitely part of the file name. We only ever use `Mathlib/Tactic/Basic.lean` to refer to the file on disk, or `Mathlib.Tactic.Basic` to refer to the module name. Let's not mix and match conventions by using `Mathlib/Tactic/Basic`.